casadm/cas_lib.c

```diff
@@ -882,6 +1056,32 @@ static void check_cache_scheduler(const char *cache_device, const char *elv_name
 	}
 }
 
+static int set_cache_path(char *path, const char *cache_dev, size_t dev_name_len)
+{
+	char abs_dev_path[MAX_STR_LEN];		// absolute path of disk
+
+	/* save given path as absolute path in temporary variable */
+	if (get_abs_path(abs_dev_path, cache_dev, MAX_STR_LEN) < 0)
+		return FAILURE;
+
+	if (_is_by_id_path(abs_dev_path)) {
+		/* attempt to get user-provided path if it's by-id path */
+		if (get_by_id_path(abs_dev_path, path, dev_name_len) == SUCCESS)
```

Comment:
why convert the path to by-id if it already is by id?

Reply:
It's the redundant check and I can drop it.